src/perftest_communication.c

```diff
@@ -2321,7 +2320,7 @@ int rdma_cm_connect_events(struct pingpong_context *ctx,
 		rc = rdma_cm_events_dispatcher(ctx, user_param, event->id, event);
 		if (rc) {
 			error_message = "Failed to handle RDMA CM event.";
-			goto error;
+			error_handler(error_message);
```

Comment:
It is not enough, you will fall back to  rdma_ack_cm_event() below and have all potential to stuck in the endless while loop if rdma_ack_cm_event returns 0.

Reply:
@rleon  You are right , I will fix this

Reply:
Done